encode.go

```diff
@@ -112,7 +113,9 @@ func (e *Encoder) Encode(v interface{}) error {
 		}
 
 	case float32:
-		e.buffer = encFloat(e.buffer, float64(val))
+		str := strconv.FormatFloat(float64(val), 'f', -1, 32)
```

Comment:
suggest to add a new func encFloat32 to receive a float32 and encode it, the string converting would degrade performance